mesonbuild/dependencies/base.py

```diff
@@ -1791,7 +1791,7 @@ def _shebang_to_cmd(script):
                 # the single argument to pass to that command. So we must split
                 # exactly once.
                 commands = first_line[2:].split('#')[0].strip().split(maxsplit=1)
-                if mesonlib.is_windows():
+                if mesonlib.is_windows() or mesonlib.is_cygwin():
                     # Windows does not have UNIX paths so remove them,
                     # but don't remove Windows paths
                     if commands[0].startswith('/'):
```

Comment:
We shouldn't strip all unix paths on Cygwin though, because Cygwin *does* understand unix paths. We need to detect at least these conditions: `#!/usr/bin/python3` and `#!/usr/bin/env python3` and `#!python3` and substitute the correct Python 3.